### PR TITLE
[POAE7-2672] Support order by/ top-n/ limit node in cider-velox

### PR DESCRIPTION
**Pull request by ZJie1**
### What changes were proposed in this pull request?

- Update velox submodule
- Support OrderBy/TopN/Limit Node in cider-velox part.

### Why are the changes needed?

new feature support.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UT.

### CODE REVIEW

**Comment by ZJie1** [comment]
@chaojun-zhang @harborn  @haiweizh Please help take a review.

File: cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp [hunk 1]
```diff
@@ -103,6 +103,31 @@ void VeloxPlanFragmentToSubstraitPlan::reconstructVeloxPlan(
         return std::make_shared<FilterNode>(
             filterNode->id(), filterNode->filter(), planBuilder_->planNode());
       });
+    } else if (auto orderByNode = std::dynamic_pointer_cast<const OrderByNode>(*riter)) {
+      planBuilder_->addNode([&](std::string id, core::PlanNodePtr input) {
```
Comment (harborn) [inline cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp:107]: seems `id` and `input` not used?
so why the lambda function should pass this two args?

File: cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp [hunk 2]
```diff
@@ -100,6 +100,122 @@ class VeloxPlanFragmentSubstraitConverterTest : public OperatorTestBase {
...
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, topNTwoKeys) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .filter("c0 > 15")
+                  .topN({"c0 NULLS FIRST", "c1 DESC NULLS LAST"}, 10, true)
+                  .planNode();
+
+  assertQuery(plan,
```
Comment (haiweizh) [inline cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp:209]: Now cider is not support order by/offset/limit, so why these query cases passed?

File: cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp [hunk 3]
```diff
@@ -100,6 +100,122 @@ class VeloxPlanFragmentSubstraitConverterTest : public OperatorTestBase {
...
+              "SELECT * FROM tmp WHERE c0 > 15 ORDER BY c0 NULLS FIRST, c1 DESC "
+              "NULLS LAST LIMIT 10");
+
+  assertPlanConversion(plan,
```
Comment (haiweizh) [inline cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp:213]: Seems plan convert case does not contain complex case, such as: order by Multicolumn、order by expression、order by 1.

**Review by haiweizh — COMMENTED** [review]
Seems this pr just convert the velox execution plan to substrait plan, how about merge and resort logic？Will support it in next pr or that is already supported？

File: cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp (hunk 3 above)
Comment (ZJie1) [reply cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp:213]: we can cover this by sql files, will add them into .sql

File: cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp (hunk 1 above)
Comment (ZJie1) [reply cider-velox/src/substrait/VeloxPlanFragmentToSubstraitPlan.cpp:107]: See here.
```
  /// Add a user-defined PlanNode as the root of the plan. 'func' takes
  /// the current root of the plan and returns the new root.
  PlanBuilder& addNode(
      std::function<core::PlanNodePtr(std::string nodeId, core::PlanNodePtr)>
          func) {
    planNode_ = func(nextPlanNodeId(), planNode_);
    return *this;
  }
```

File: cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp [hunk 4]
```diff
@@ -100,6 +100,122 @@ class VeloxPlanFragmentSubstraitConverterTest : public OperatorTestBase {
       std::make_shared<SubstraitVeloxPlanConverter>(pool_.get());
 };
 
+TEST_F(VeloxPlanFragmentSubstraitConverterTest, orderBySingleKey) {
+  auto vectors = makeVector(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan =
+      PlanBuilder().values(vectors).orderBy({"c0 DESC NULLS LAST"}, false).planNode();
+  assertQuery(plan, "SELECT * FROM tmp ORDER BY c0 DESC NULLS LAST");
```
[the remark this answers is not in the document]
Comment (ZJie1) [reply cider-velox/test/SubstraitTest/VeloxPlanFragmentSubstraitConverterTest.cpp:108]: Per discussed offline, let's sync this again after cider support the orderby node.

**Review by haiweizh — APPROVED** [review]

**Review by harborn — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**